core/include/vecmem/memory/unique_ptr.hpp

```diff
@@ -0,0 +1,462 @@
+#include <type_traits>
+
+#include "vecmem/memory/memory_resource.hpp"
+
+namespace vecmem {
+namespace {
```

Comment:
I don't think that an unnamed namespace would do much good in a header file. :thinking: I don't think I've seen such a setup in a public header before... I would prefer to put these types into the `vecmem::details::` namespace. As we've done for a number of other types already.

Reply:
It does pretty much what you would expect: it creates a translation-unit-local anonymous namespace, except it's still inside of the `vecmem`. But sure, I can move them to `vecmem::details` no problem. :smile: